src/scheduler/simulate.c

```diff
@@ -621,27 +621,30 @@ perform_event(status *policy, timed_event *event)
 int
 exists_run_event(event_list *calendar, time_t end)
 {
-	timed_event *te;
-
-	if (calendar == NULL)
-		return 0;
-
-	te = get_next_event(calendar);
-
-	if (te == NULL) /* no events in our calendar */
-		return 0;
-
-	te = find_init_timed_event(te, IGNORE_DISABLED_EVENTS, TIMED_RUN_EVENT);
-
-	if (te == NULL) /* no run event */
+	if (calendar == NULL || calendar->first_run_event == NULL)
 		return 0;
+	
+	if (calendar->first_run_event->event_time < end)
+		return 1;
+	return 0;
+}
 
-	/* there is a run event, but it's after end */
-	if (end != 0 && te->event_time > end)
+/**
+ * @brief
+ * 		returns 1 if there is a run event before the end time on a node
+ * @param[in]	ninfo - the node to check for
+ * @param[in]	search between now and end
+ */
+int
+exists_run_event_on_node(node_info *ninfo, time_t end)
+{
+	if (ninfo == NULL || ninfo->node_events == NULL)
 		return 0;
+	
+	if (ninfo->node_events->event->event_time < end)
```

Comment:
shouldn't you check here if 'event' is a run event or not?

Reply:
and will there only ever be a single event in 'node_events' ?

Reply:
node_events only contains run events.  Since it contains all the run events, it is possible there is only one.  I'm not sure I understand the question.

Reply:
ah, if it only contains end events then it makes sense that you just check for one. Can we rename the variable to "run_events" so that it's clear that these are expected to be only run events?

Reply:
I'd rather not rename it.  I named it node_events because I wasn't sure if we were going to expand it in the future to contain all events on that node.

Reply:
Ok, can you please add a comment here then that says that node_events only contains run events so just check the first one?

Reply:
I just changed the comment for the member to say that.